mantidproject/mantid
# Add unit tests for instrument viewer

**Description of work.**

- This PR adds unit tests for calls to the SimpleWidget and MantidGLWidget members of instrument viewer. 
- It adds a class that allows us to mock Qt connect calls and adds tests for these too.

**To test:**

- Load some data, open Instrument Viewer and play around with the different views. Ensure everything is working as expected.
- Toggle the GL option in the Mantid settings and re-test.

Fixes #31763

*This does not require release notes* because **it concerns unit tests only**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
